app/models/user.rb

```diff
@@ -3,16 +3,52 @@ class User < ActiveRecord::Base
     config.authentications_class = Authentication
   end
 
+  after_create :add_current_deck
+  belongs_to :current_deck, class_name: "Deck"
+  has_many :decks
   has_many :authentications, dependent: :destroy
-  accepts_nested_attributes_for :authentications
-
-  validates :password, length: { minimum: 3 }
-  validates :password, confirmation: true
-  validates :password_confirmation, presence: true
+  has_many :cards
 
+  accepts_nested_attributes_for :authentications
+  with_options if: :password_present? do |u|
+    u.validates :password, length: { minimum: 3 }, on: [:create, :update]
+    u.validates :password, confirmation: true, on: [:create, :update]
+    u.validates :password_confirmation, presence: true, on: [:create, :update]
+  end
   validates :email, uniqueness: true, presence: true, on: :create, email: true
   validates :email, email: true, on: :update
-  has_many :cards
+
+  def add_current_deck
+    decks.create(name: "Default deck")
+  end
+
+  def current_deck_or_any
```

Comment:
Assignment Branch Condition size for current_deck_or_any is too high. [15.33/15]